tests/functional/tests/engine/test_partial_hit.py

```diff
@@ -0,0 +1,205 @@
+#
+# Copyright(c) 2024 Huawei Technologies
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+import pytest
+
+
+from pyocf.types.data import Data, DataSeek
+from pyocf.types.cache import Cache, CacheMode
+from pyocf.types.core import Core
+from pyocf.types.shared import CacheLineSize
+from pyocf.types.volume import RamVolume, Volume
+from pyocf.types.volume_core import CoreVolume
+from pyocf.utils import Size
+from pyocf.types.io import WriteMode, IoDir
+
+
+def assert_device(device: Volume, reads, writes):
+    stats = device.get_stats()
+    if reads is not None:
+        assert (
+            stats[IoDir.READ] == reads
+        ), f"actual reads: {stats[IoDir.READ]}, expected reads: {reads}"
+    if writes is not None:
+        assert (
+            stats[IoDir.WRITE] == writes
+        ), f"actual writes: {stats[IoDir.WRITE]}, expected writes: {writes}"
+
+
+@pytest.mark.parametrize("cacheline_size", CacheLineSize)
+@pytest.mark.parametrize("cache_mode", CacheMode)
+def test_partial_hit_write(pyocf_ctx, cacheline_size, cache_mode):
+    cache_device = RamVolume(Size.from_MiB(50))
+    core_device = RamVolume(Size.from_MiB(50))
+
+    cache = Cache.start_on_device(
+        cache_device, cache_line_size=cacheline_size, cache_mode=cache_mode
+    )
+    core = Core.using_device(core_device)
+
+    queue = cache.get_default_queue()
+
+    cache.add_core(core)
+    core_volume = CoreVolume(core)
+    core_volume.open()
+
+    # Fill core with data
+    CL = cache.cache_line_size
+    data = Data(CL // 2)
+    data.seek(DataSeek.BEGIN, 0)
+    data.write(b"A\x00\x00\x00\x00", 5)
+    core_device.sync_io(queue, 0, data, IoDir.WRITE, 0, 0)
```

Comment:
Here as well no need for "io_class" and "flags" arguments of sync_io() in several places.

Reply:
done